core/src/test/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackendSuite.scala

```diff
@@ -184,4 +184,52 @@ class CoarseMesosSchedulerBackendSuite extends SparkFunSuite
 
     verify(driver, times(1)).reviveOffers()
   }
+
+  test("testing various isOfferValidForScheduling cases") {
```

Comment:
Ideally I would like to see each test case tests one condition as it's hard to know what's going on here.
Can you at least organized them to related ones (constraints, required mem/cpu, etc).